From 7e2bad6bf74d91f9cd876a4620796fe4eb5f3514 Mon Sep 17 00:00:00 2001
From: Xiyue Deng <manphiz@gmail.com>
Date: Thu, 4 Jul 2024 02:16:52 -0700
Subject: [PATCH 1/4] Don't recursively add elpa path to match package.el
 behavior

* Add .nosearch to `/usr/share/emacs/site-lisp/elpa{,-src}' using
triggers in dh-elpa, which will disable subdirs.el from processing
those directories.
---
 debian/install     | 2 ++
 elpa-src/.nosearch | 0
 elpa/.nosearch     | 0
 3 files changed, 2 insertions(+)
 create mode 100644 elpa-src/.nosearch
 create mode 100644 elpa/.nosearch

diff --git a/debian/install b/debian/install
index d6e2470..64fc1fc 100644
--- a/debian/install
+++ b/debian/install
@@ -2,3 +2,5 @@ elpa.pm usr/share/perl5/Debian/Debhelper/Sequence
 emacsen-common usr/share/debhelper/dh_elpa
 autoscripts/prerm-elpa usr/share/debhelper/autoscripts
 usr/bin
+elpa/.nosearch usr/share/emacs/site-lisp/elpa
+elpa-src/.nosearch usr/share/emacs/site-lisp/elpa-src
diff --git a/elpa-src/.nosearch b/elpa-src/.nosearch
new file mode 100644
index 0000000..e69de29
diff --git a/elpa/.nosearch b/elpa/.nosearch
new file mode 100644
index 0000000..e69de29
-- 
2.39.2

